content/influxdb/v2.0/api-guide/client-libraries/nodejs/query.md

```diff
@@ -56,24 +56,20 @@ The following example sends a Flux query to an InfluxDB bucket and outputs rows
    ```
    Replace *`YOUR_BUCKET`* with the name of your InfluxDB bucket.
 
-4. Use the `queryRows()` method of the query client to query InfluxDB.
-   `queryRows()` takes a Flux query and an [RxJS **Observer**](http://reactivex.io/rxjs/manual/overview.html#observer) object.
-   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an  [RxJS **Observable**](http://reactivex.io/rxjs/manual/overview.html#observable).
-   `queryRows()` subscribes your observer to the observable.
-   Finally, the observer logs the rows from the response to the terminal.
+4. Use the `iterateRows()` method of the query client to query InfluxDB.
+   `iterateRows()` takes a Flux query and returns table as an asynchronous collection.
+   The client returns [table](/{{% latest "influxdb" %}}/reference/syntax/annotated-csv/#tables) metadata and rows as an as an AsyncIterable.
 
    ```js
-   const observer = {
-     next(row, tableMeta) {
-       const o = tableMeta.toObject(row)
+   const myQuery = async () => {
+     for await (const {values, tableMeta} of queryApi.iterateRows(fluxQuery)) {
+       const o = tableMeta.toObject(values)
        console.log(
          `${o._time} ${o._measurement} in '${o.location}' (${o.sensor_id}): ${o._field}=${o._value}`
        )
      }
    }
-
-   queryApi.queryRows(fluxQuery, observer)
-
+   myQuery()
    ```
```

Comment:
```suggestion
   ```

To learn more about `for await...of` and async iterable objects, see the [MDN `for await...of` documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of).
```